### ISSUE TITLE
Management zone creation not totally completed by the provider before deploying depending resources such as key requests

### ISSUE THREAD

**Issue by ftmazzone** [dynatrace-oss/terraform-provider-dynatrace#519]
**Describe the bug**
When creating a new management zone with the resource [dynatrace_management_zone_v2 ](https://registry.terraform.io/providers/dynatrace-oss/dynatrace/latest/docs/resources/key_requests), Terraform does not wait for the end of the management zone creation before deploying the resources depending on this management zone. 

When referencing the management zone to create a key request [dynatrace_key_requests](https://registry.terraform.io/providers/dynatrace-oss/dynatrace/latest/docs/resources/key_requests), the deployment will fail as the management zone is still not available.
Retrying the same deployment after a few seconds delay will work as the management zone is then ready.

**To Reproduce**
1. Deploy the following ressources in two Terraform modules
```
# In a module "management_zones" to create the management zone
resource "dynatrace_management_zone_v2" "test" {
  name        = "test_${var.customerName}_${var.environmentType}"
  description = "Kubernetes cluster dynakube-${var.customerName}${var.environmentType}"
  rules 
    rule {
      type            = "SELECTOR"
      enabled         = true
      entity_selector =<<-EOT
        type("AWS_LAMBDA_FUNCTION"),tag("[AWS]Environment:${var.customerName}${var.environmentType}")
      EOT
    }
  }

# In a module "key_requests" to create the key request with `depends_on=management_zones`

data "dynatrace_management_zone_v2" "test" {
  name = "test"
}

data "dynatrace_entity" "test" {
  entity_selector = <<-EOT
  mzName("${data.dynatrace_management_zone_v2.test.name}"),type(service),entityName.equals("test - * ")
  EOT
}

resource "dynatrace_key_requests" "test" {
  names   = ["/test1", "/test2"]
  service = data.dynatrace_entity.test.id
}
```
3. Create the Terraform plan
4. Apply the plan
5. See error `Error: Missing required argument [...]
The argument "service" is required, but no definition was found.`
6. Update the Terraform plan
7. Apply the plan
8. Deployment is successful

**Expected behavior**
The management zone creation should be fully completed before creating the resources depending from the management zone.

**Screenshots**
[N/A]

**Desktop (please complete the following information):**
 - OS: Windows 
 - Version: Terraform v1.9.4 on windows_amd64

**Additional context**
As a workaround, the module [time_sleep](https://registry.terraform.io/providers/hashicorp/time/latest/docs/resources/sleep) with `create_duration="15s"` allows to wait a few seconds before creating the key request. The deployment will then be successful without retry.

```
resource "time_sleep" "wait_management_zones_creation" {
depends_on      = [module.management_zones]
create_duration = "15s"
 }
```

**Comment by kishikawa12** [comment]
Hi @ftmazzone,

Thank you for raising the issue. Unfortunately it's been a constant battle with certain resources as they could have a delay to propagate the configuration across all cluster nodes.

A polling mechanism already exists to validate the creation but certain workflows could end up taking even longer. Rather than increasing the default values, I've introduced a couple of environment variables that can be used to fine tune these settings.

- `DT_MGMZ_RETRIES` (Default: 50, Max: 150) configures the maximum attempts to confirm that the create operation has succeeded.
- `DT_MGMZ_SUCCESSES` (Default: 5, Max: 25) configures the number of successful consecutive retries expected.

As reference, I was able to replicate your issue and was able to resolve it by setting `DT_MGMZ_SUCCESSES` to 10 and `DT_MGMZ_RETRIES` to 80.

This feature/fix will be included in the release planned for later this week.
